routing/localchans/manager.go

```diff
@@ -212,6 +212,14 @@ func (r *Manager) updateEdge(tx kvdb.RTx, chanPoint wire.OutPoint,
 		edge.MinHTLC = *newSchema.MinHTLC
 	}
 
+	// If the channel is unannounced we make sure the dont_forward bit is
+	// set in the ChannelEdgePolicy. This is an on-the-fly migration. The
+	// edge is persisted to disk later when propagating the new channel
+	// update.
+	if edgeInfo.AuthProof == nil {
+		edge.MessageFlags |= lnwire.ChanUpdateDontForward
+	}
```

Comment:
Good Catch, I need to add a testcase somewhere where we switch from unannounced to announced (zero-conf public channel). Because the current code would just have add the same edgeinfo as before including the msg flag (retrieving it from the db).
 
Lnd makes sure that we only have a valid proof specified when the channel was announced to the network so I think it is even stricter than just looking at the `dont_forward` bit.

Reply:
Yeah judging by some of the other parts of this PR, it seems that the `AuthProof` is a *necessary but not sufficient condition` for forwarding. We must also have an *absence* of the `dont_forward` bit. I think we need to mentally decouple the idea of the presence of the `AuthProof` and the decision to forward. I do not believe that zero-conf public channels will be the last time we see these ideas differ from one another.